File: chapter5/typescript/PizzaRequest.ts

/**
 * Pizza request sending from client request
 *
 * @Author Bridget Wu
 */
 export class PizzaRequest {


    private sauces: String[] = [];

    private toppings: String[] = [];

    public getSauces() : String[]{
        return this.sauces;
    }

    /**
     * add a sauce
     *
     * @param sauce
     */
    public addSauce(sauce: String): void {
        this.sauces.push(sauce);
    }

    public getToppings(): String[] {
        return this.toppings;
    }

    /**
     * add a topping
     *
     * @param topping
     */
    public addTopping(topping: String): void {
        this.toppings.push(topping);
    }
}

File: chapter5/java/designpattern/src/main/java/com/designpattern/PizzaRequest.java
package com.designpattern;

import java.util.ArrayList;
import java.util.List;

/**
 * Pizza request sending from client request
 *
 * @Author Bridget Wu
 */
public class PizzaRequest {

    List<String> sauces = new ArrayList<>();

    List<String> toppings = new ArrayList<>();

    public List<String> getSauces() {
        return sauces;
    }

    /**
     * add a sauce
     *
     * @param sauce
     */
    public void addSauce(String sauce) {
        this.sauces.add(sauce);
    }

    public List<String> getToppings() {
        return toppings;
    }

    /**
     * add a topping
     *
     * @param topping
     */
    public void addTopping(String topping) {
        this.toppings.add(topping);
    }
}
